.github/workflows/cmake.yml

```diff
@@ -65,27 +102,87 @@ jobs:
             compiler: gcc-11
             cflags: -Werror -Wall -Wextra
 
+          - name: Windows OW DOS 16-bit
+            os: windows-latest
+            compiler: watcom
+            cflags: -ml
+            cmake-args: -G "Watcom WMake" -D CMAKE_SYSTEM_NAME=DOS -D CMAKE_SYSTEM_PROCESSOR=I86
+            artifact: windows-dos-16
+
+          - name: Windows OW DOS 32-bit
+            os: windows-latest
+            compiler: watcom
+            cmake-args: -G "Watcom WMake" -D CMAKE_SYSTEM_NAME=DOS -D CMAKE_SYSTEM_PROCESSOR=X86
+            artifact: windows-dos-32
+
+          - name: Windows OW OS/2 16-bit
+            os: windows-latest
+            compiler: watcom
+            cflags: -ml
+            cmake-args: -G "Watcom WMake" -D CMAKE_SYSTEM_NAME=OS2 -D CMAKE_SYSTEM_PROCESSOR=I86
+            artifact: windows-os2-16
+
+          - name: Windows OW OS/2 32-bit
+            os: windows-latest
+            compiler: watcom
+            cmake-args: -G "Watcom WMake" -D CMAKE_SYSTEM_NAME=OS2 -D CMAKE_SYSTEM_PROCESSOR=X86
+            artifact: windows-os2-32
+
+          - name: Windows OW Windows 32-bit
+            os: windows-latest
+            compiler: watcom
+            watcom-test: 1
+            cmake-args: -G "Watcom WMake"
+
+          - name: Windows OW Linux 32-bit
+            os: windows-latest
+            compiler: watcom
+            cmake-args: -G "Watcom WMake" -D CMAKE_SYSTEM_NAME=Linux -D CMAKE_SYSTEM_PROCESSOR=X86
+
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
 
+    - name: Install packages (Open Watcom)
+      if: matrix.compiler == 'watcom'
+      uses: open-watcom/setup-watcom@v0
     - name: Install packages (Windows)
-      if: runner.os == 'Windows'
+      if: runner.os == 'Windows' && matrix.compiler != 'watcom'
       run: |
         choco install --no-progress ninja ${{ matrix.packages }}
 
+    - name: Generate project files (Open Watcom)
+      if: matrix.compiler == 'watcom'
+      run: cmake -S ${{ matrix.src-dir || '.' }} -B ${{ matrix.build-dir || 'build' }} ${{ matrix.cmake-args }} -D CMAKE_BUILD_TYPE=${{ matrix.build-config || 'Release' }}
+      env:
+        CFLAGS: ${{ matrix.cflags }}
     - name: Generate project files
-      run: cmake -S ${{ matrix.src-dir || '.' }} -B ${{ matrix.build-dir || '.' }} ${{ matrix.cmake-args }} -D CMAKE_BUILD_TYPE=${{ matrix.build-config || 'Release' }}
+      if: matrix.compiler != 'watcom'
```

Comment:
This step shouldn't need a separate version for watcom.

Reply:
I don't want to change anything for other compilers and matrix.compiler is as class for watcom instead of for others. Also cmake support Watcom compilers cross-compilation without need of special cmake configuration scripts. it is control directly by CMAKE_SYSTEM_NAME and CMAKE_SYSTEM_PROCESSOR macros. And also reference to current directory for output files is not good solution for Watcom because some intermediate files created by Watcom overwrite zlib source files. That during testing destroys sources and you cannot repeate steps. I did it this simple way that matrix.compiler is use to switch to watcom logic and not control Watcom version etc.

Reply:
I separated CI CMake build change from OS/2 build fix that here is only CI CMake related changes

Reply:
NOTE for CMake use with watcom:
- watcom must be used with -G "Watcom WMake" because it setup WATCOM stuff in CMake
- watcom mode in CMake doesn't use POSIX CC mechanism, therefore you cannot use CC environment variable